libraries/stdlib/jvm/test/random/RandomSerializationTest.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package random
+
+import test.io.deserializeFromHex
+import test.io.serializeAndDeserialize
+import kotlin.random.Random
+import kotlin.random.asJavaRandom
+import kotlin.random.asKotlinRandom
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertSame
+
+class RandomSerializationTest {
+    @Test
+    fun defaultIsSerializable() {
+        val instance = Random
+        assertSame(instance, serializeAndDeserialize(instance))
+    }
+
+    private fun testPersistedDeserialization(hexValue: String, expected: Any) =
+        assertEquals(expected = expected, actual = deserializeFromHex(hexValue))
+
+    @Test
+    fun deserializeDefault() = testPersistedDeserialization(
+        "ac ed 00 05 73 72 00 1c 6b 6f 74 6c 69 6e 2e 72 61 6e 64 6f 6d 2e 52 61 6e 64 6f 6d 24 44 65 66 61 75 6c 74 59 81 49 e9 14 4e a8 28 02 00 00 78 70",
+        Random
+    )
+
+    @Test
+    fun xorwowIsSerializable() {
+        val instance = Random(0)
+        val deserialized = serializeAndDeserialize(instance)
+        assertEquals(instance.nextInt(), deserialized.nextInt())
```

Comment:
@ilya-g of course, we can check more than first 4 bytes the deserialized random produces.